nixos/modules/services/misc/tiddlywiki.nix

```diff
@@ -0,0 +1,82 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.tiddlywiki;
+  listenParams = concatStrings (mapAttrsToList (n: v: " '${n}=${toString v}' ") cfg.listenOptions);
+  package = pkgs.nodePackages.tiddlywiki;
+  exe = "${package}/lib/node_modules/.bin/tiddlywiki";
+
+in {
+
+  options.services.tiddlywiki = {
+
+    enable = mkEnableOption "TiddlyWiki nodejs server";
+
+    dataDir = mkOption {
+      type = types.path;
+      default = "/var/lib/tiddlywiki";
+      description = "Path to wiki folder.";
+    };
+
+    listenOptions = mkOption {
+      type = types.attrs;
+      default = {};
+      example = {
+        credentials = "../credentials.csv";
+        readers="(authenticated)";
+        port = 3456;
+      };
+      description = ''
+        Parameters passed to <literal>--listen</literal> command.
+        Refer to <link xlink:href="https://tiddlywiki.com/#WebServer"/>
+        for details on supported values.
+      '';
+    };
+
+    user = mkOption {
+      type = types.str;
+      default = "tiddlywiki";
+      description = "User account under which TiddlyWiki runs.";
+    };
+
+    group = mkOption {
+      type = types.str;
+      default = "tiddlywiki";
+      description = "Group under which TiddlyWiki runs.";
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd = {
+      services.tiddlywiki = {
+        description = "TiddlyWiki nodejs server";
+        after = [ "network.target" "systemd-tmpfiles-setup.service" ];
```

Comment:
The dependency on `systemd-tmpfiles-setup.service` **shouldn't** be necessary... I can't find relevant PR with discussion at the moment.

Reply:
Agree, removed it